app/models/scenario_scaling.rb

```diff
@@ -24,6 +24,14 @@ def self.from_atlas_scaling(scaling)
     new(scaling.attributes)
   end
 
+  def self.from_scenario(scenario)
+    if scenario.area['derived']
```

Comment:
The area is a `HashWithIndifferentAccess`, so you can use Symbols for keys instead of allocating extra strings (same applies when Scenario reads `scaling`).